synapse/handlers/sliding_sync/room_lists.py

```diff
@@ -198,6 +201,309 @@ async def compute_interested_rooms(
+
+        # Handle state resets in the from -> to token range.
+        state_reset_rooms = (
+            newly_left_room_map.keys() - room_membership_for_user_map.keys()
+        )
+        if state_reset_rooms:
+            room_membership_for_user_map = dict(room_membership_for_user_map)
+            for room_id in (
+                newly_left_room_map.keys() - room_membership_for_user_map.keys()
+            ):
+                # Get the state at the time. Note that room type never changes,
+                # so we can just get current room type
+                room_type = await self.store.get_room_type(room_id)
+                is_encrypted = await self.get_is_encrypted_for_room_at_token(
+                    room_id, newly_left_room_map[room_id].to_room_stream_token()
+                )
+
+                room_membership_for_user_map[room_id] = RoomsForUserSlidingSync(
+                    room_id=room_id,
+                    sender=None,
+                    membership=Membership.LEAVE,
+                    event_id=None,
+                    event_pos=newly_left_room_map[room_id],
+                    room_version_id=await self.store.get_room_version_id(room_id),
+                    room_type=room_type,
+                    is_encrypted=is_encrypted,
+                )
+
+        if sync_config.lists:
+            sync_room_map = {
+                room_id: room_membership_for_user
+                for room_id, room_membership_for_user in room_membership_for_user_map.items()
+                if filter_membership_for_sync(
+                    user_id=user_id,
+                    room_membership_for_user=room_membership_for_user,
+                    newly_left=room_id in newly_left_room_map,
+                )
+            }
+            with start_active_span("assemble_sliding_window_lists"):
+                for list_key, list_config in sync_config.lists.items():
+                    # Apply filters
+                    filtered_sync_room_map = sync_room_map
+                    if list_config.filters is not None:
+                        filtered_sync_room_map = await self.filter_rooms_using_tables(
+                            user_id,
+                            sync_room_map,
+                            list_config.filters,
+                            to_token,
+                            dm_room_ids,
+                        )
+
+                    # Find which rooms are partially stated and may need to be filtered out
+                    # depending on the `required_state` requested (see below).
+                    partial_state_room_map = (
+                        await self.store.is_partial_state_room_batched(
+                            filtered_sync_room_map.keys()
+                        )
+                    )
+
+                    # Since creating the `RoomSyncConfig` takes some work, let's just do it
+                    # once and make a copy whenever we need it.
```

Comment:
Prior art on this comment but we should update because `RoomSyncConfig` is frozen and we don't need to take copies anymore.

```suggestion
                    # Since creating the `RoomSyncConfig` takes some work, let's just do it
                    # once.
```

(in both places)